src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -34,6 +35,15 @@
 from azure.cli.command_modules.network.zone_file.parse_zone_file import parse_zone_file
 from azure.cli.command_modules.network.zone_file.make_zone_file import make_zone_file
 
+logger = azlogging.get_az_logger(__name__)
+
+def _upsert(collection, obj, key_name, key_value):
+    match = next((x for x in collection if getattr(x, key_name) == key_value), None)
```

Comment:
The `getattr` could through an exception, you may want to give a default value to prevent that. since the `next` method already is given a default value, I assume exception is not ideal here.